odl/tomo/geometry/geometry.py

```diff
@@ -39,10 +38,14 @@ class Geometry(with_metaclass(ABCMeta, object)):
       parameters,
     * a mapping from the motion and surface parameters to the detector pixel
       direction to the source,
-    * optionally a mapping from the motion parameters to the source position
+    * optionally a mapping from the motion parameters to the source position,
+    * optionally a global translation of the geometry (shift of the origin)
+
+    For details, check `the online docs
+    <https://odlgroup.github.io/odl/guide/geometry_guide.html>`_.
     """
 
-    def __init__(self, ndim, motion_part, detector):
+    def __init__(self, ndim, motion_part, detector, translation=None):
```

Comment:
I mainly have shifted rotation axes in mind. So an option would be to do this only in `AxisOrientedGeometry` and add an optional `axis_shift` there.

But frankly I added it since [you asked for it](https://github.com/odlgroup/odl/pull/968#discussion_r113924206) :-)